### PR TITLE
Allow setting HTTP headers and parameters during schema upload

### PR DESCRIPTION
**Pull request by yupeng9**
## Description
Useful for sender to annotate additional information for observability and auditing

### CODE REVIEW

File: pinot-common/src/main/java/org/apache/pinot/common/utils/FileUploadDownloadClient.java [hunk 1]
```diff
@@ -243,8 +243,9 @@ private static HttpUriRequest constructDeleteRequest(URI uri) {
     return requestBuilder.build();
   }
 
-  private static HttpUriRequest getAddSchemaRequest(URI uri, String schemaName, File schemaFile) {
-    return getUploadFileRequest(HttpPost.METHOD_NAME, uri, getContentBody(schemaName, schemaFile), null, null,
+  private static HttpUriRequest getAddSchemaRequest(URI uri, String schemaName, File schemaFile, List<Header> headers,
```
Comment (Jackie-Jiang) [inline pinot-common/src/main/java/org/apache/pinot/common/utils/FileUploadDownloadClient.java:246]: Add nullable annotation to headers and parameters

File: pinot-common/src/main/java/org/apache/pinot/common/utils/FileUploadDownloadClient.java [hunk 2]
```diff
@@ -419,7 +420,25 @@ public SimpleHttpResponse sendDeleteRequest(URI uri)
...
+   * @param uri URI
+   * @param schemaName Schema name
+   * @param schemaFile Schema file
+   * @param headers HTTP headers
+   * @param parameters HTTP parameters
+   * @return Response
+   * @throws IOException
+   * @throws HttpErrorStatusException
+   */
+  public SimpleHttpResponse addSchema(URI uri, String schemaName, File schemaFile, List<Header> headers,
```
Comment (Jackie-Jiang) [inline pinot-common/src/main/java/org/apache/pinot/common/utils/FileUploadDownloadClient.java:438]: Add nullable annotation to headers and parameters

**Review by Jackie-Jiang — APPROVED** [review]
LGTM with minor comments

**Comment by yupeng9** [comment]
@Jackie-Jiang Thanks for the review. Comments addressed.

**Pull request merged**